linera-core/src/client/mod.rs

```diff
@@ -938,12 +925,30 @@ where
         #[cfg(with_metrics)]
         let _latency = metrics::PREPARE_CHAIN_LATENCY.measure_latency();
 
-        // Verify that our local storage contains enough history compared to the
-        // expected block height. Otherwise, download the missing history from the
-        // network.
-        let next_block_height = self.next_block_height();
+        let mut info = self.sync_until(self.next_block_height()).await?;
+
+        // If, during sync above, we've learned about new owners we need to check if the chain
+        // has been extended beyoned `self.next_block_height()`.
```

Comment:
Even if we didn't just learn about them, and even if they've been co-owners since the chain was created: As long as we don't know we are the only ones who can add blocks, we have to do this check.

Reply:
Maybe just write "If we are not the only owners". It's not only about the case where we just learned about the others. But maybe the existing comment below is already enough?

Reply:
I'll remove these two lines entirely.